### PR TITLE
feat(dot): add bashls and shellcheck

### PR DESCRIPTION
**Pull request by rubiin**
I work with shell files a lot , so this basically adds support for shell files

### CODE REVIEW

File: lua/lazyvim/plugins/extras/lang/shell.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      servers = {
+        bashls = {},
+      },
+    },
+  },
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "bash" })
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/shell.lua:14]: This is already included in LazyVim

File: lua/lazyvim/plugins/extras/lang/shell.lua [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      end
+    end,
+  },
+  {
+    "williamboman/mason.nvim",
+    opts = function(_, opts)
+      vim.list_extend(opts.ensure_installed or {}, {
+        -- formatters & linters
+        "shfmt",
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/shell.lua:23]: shfmt is not needed with bashls

**Review by folke — CHANGES_REQUESTED** [review]
We already have an extra called `dot` for this kind of stuff.
Would be great if you can add bashls and shellcheck in there.

File: lua/lazyvim/plugins/extras/lang/shell.lua (hunk 2 above)
Comment (folke) [reply lua/lazyvim/plugins/extras/lang/shell.lua:23]: Scrap that. It **is** needed and I'm actually already using that for formatting :)

File: lua/lazyvim/plugins/extras/lang/shell.lua (hunk 1 above)
Comment (rubiin) [reply lua/lazyvim/plugins/extras/lang/shell.lua:14]: sure will do that instead

File: lua/lazyvim/plugins/extras/lang/shell.lua (hunk 2 above)
Comment (rubiin) [reply lua/lazyvim/plugins/extras/lang/shell.lua]: Moved the changes to dot file instead. I think I also need to change the PR title

**Comment by folke** [comment]
Thanks!

**Pull request merged**